client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/HashBasedShuffleWriter.java

```diff
@@ -174,14 +174,12 @@ public void write(scala.collection.Iterator<Product2<K, V>> records) throws IOEx
       } else {
         write0(records);
       }
+      close();
     } catch (InterruptedException e) {
+      abort();
       TaskInterruptedHelper.throwTaskKillException();
-    } finally {
-      try {
-        close();
-      } catch (InterruptedException e) {
-        TaskInterruptedHelper.throwTaskKillException();
-      }
+    } catch (Exception e) {
+      abort();
     }
```

Comment:
Instead, here and below, why not use a pattern like:
```
var needAbort = true
try {
  <code>
  needAbort = false
} 
// other catch blocks, which might set needAbort = false
finally {
  if (needAbort) {
    abort()
  }
}

```

(If we need `throwTaskKillException` in `abort`, we will need to modify to handle that case as well)